pkg/database/db_factory.go

```diff
@@ -0,0 +1,300 @@
+package database
+
+import (
+	"database/sql"
+	"errors"
+	"fmt"
+	glog "log"
+	"net/url"
+	"os"
+	"regexp"
+	"strings"
+	"time"
+
+	"github.com/rotisserie/eris"
+	"golang.org/x/exp/slices"
+
+	lru "github.com/hashicorp/golang-lru/v2"
+	"github.com/mattn/go-sqlite3"
+	log "github.com/sirupsen/logrus"
+	"gorm.io/driver/postgres"
+	"gorm.io/driver/sqlite"
+	"gorm.io/gorm"
+	"gorm.io/gorm/logger"
+	"gorm.io/plugin/dbresolver"
+)
+
+const (
+	SQLiteCustomDriverName = "sqlite3_custom_driver"
+)
+
+// dbFactory is the interface for all datasource creation.
+type dbFactory interface {
+	makeDbInstance() (*DbInstance, error)
+}
+
+// baseDbFactory are the common attributes for all datasources.
+type baseDbFactory struct {
+	dsnURL        url.URL
+	slowThreshold time.Duration
+	poolMax       int
+	reset         bool
+}
+
+// postgresDbFactory will make Postgres DbInstance.
+type postgresDbFactory struct {
+	baseDbFactory
+}
+
+// sqliteDbFactory will make Sqlite DbInstance.
+type sqliteDbFactory struct {
+	baseDbFactory
+}
+
+// ConnectDB will establish and return a DbInstance while also caching it in the global
+// var database.DB.
+func ConnectDB(dsn string, slowThreshold time.Duration, poolMax int, reset bool, migrate bool, artifactRoot string,
```

Comment:
https://github.com/G-Research/fasttrackml/issues/273

Reply:
but you can remove it right now and still stay global variable?

Reply:
if i remove it, then some other part of code will need to set the global variable

Reply:
ok lets use this one and remove another one.

Reply:
you can remove anyone and set this global in other function.we just don't need to have a wrapper on another function.

Reply:
i think this is a pretty standard pattern -- to add functionality by wrapping in another function which delegates for the shared portion. I think this is a good solution for the problem while we await refactor.

Reply:
why it can't be combined in one function? I think this is a good solution for the problem while we await refactor. - emmmm, but this task is about refactoring, so why we can do it right now? If you want I can setup everything pretty quickly without any problem.

Reply:
because smaller PRs are better

Reply:
and because, in this case, import command needs one behavior and server command needs another.

Reply:
the import function does not work when the global pointer is created, because then the second datasource clobbers the first.

Reply:
butt, `import` cmd is just another application, like `server.go`. why in `cmd/import.go` you can create 2 instances like:
```
input := NewDatabaseFactory(sqliteDSN, options)
output := NewdatabaseFactory(postgresDSN, options)
```
so no global variables, just local ones - which is what we really need. less connections with globabl `DB` instance, more incapsulated implementation, etc.